### PR TITLE
DevDocs: Link to Gridicons and Social Logos and add basic usage

### PR DESCRIPTION
**Pull request by danhauk**
This addresses #24106 by updating the icons documentation to include a link to the Gridicons and Social Logos repositories. I've also added some basic usage instructions that were missing to make implementation easier.

#### Testing

* Check out this branch `update/devdocs-iconography`.
* Confirm link to the gridicons repo and usage has been added.
* Confirm information about Social Logos has been added.

### CODE REVIEW

File: docs/icons.md [hunk 1]
```diff
@@ -4,10 +4,28 @@ This document will cover how to use icons in Calypso, as well as how to create i
 
 ## Gridicons
 
-Gridicons is the brand new icon set designed from the bottom up for Calypso. It features a consistent style.
+[Gridicons](https://github.com/automattic/gridicons) is the brand new icon set designed from the bottom up for Calypso. It features a consistent style. There's a gallery with all the available icons in https://automattic.github.io/gridicons/.
```
Comment (drw158) [inline docs/icons.md]: I would just remove the sentence about it being the "brand new icon set" and the one about consistency being a "feature" :)

**Comment by drw158** [comment]
It doesn't look like any info about SocialLogos has been added.

**Comment by danhauk** [comment]
@drw158 I forgot to commit! Just did, pull the latest to check it out.

File: docs/icons.md (hunk 1 above)
Comment (danhauk) [reply docs/icons.md]: @drw158 Thanks, I agree. I just pushed up a copy change to that section.

**Pull request merged**